packages/terminal/src/browser/terminal-widget.ts

```diff
@@ -105,32 +107,23 @@ export class TerminalWidget extends BaseWidget {
             return;
         }
 
-        this.terminalWatcher.onTerminalError((event: IBaseTerminalErrorEvent) => {
-            if (event.terminalId === this.terminalId) {
-                this.title.label = "<terminal error>";
-            }
-        });
+        this.monitorTerminal(this.terminalId);
+    }
 
-        this.terminalWatcher.onTerminalExit((event: IBaseTerminalExitEvent) => {
-            if (event.terminalId === this.terminalId) {
-                this.title.label = "<terminated>";
-            }
-        });
+    // Attach a terminal widget to an already-running shell terminal
+    public async attach(id: number): Promise<void> {
+        this.terminalId = id;
+        let success: boolean;
+        success = await this.shellTerminalServer.attach(id);
 
-        const socket = this.createWebSocket(this.terminalId.toString());
-        socket.onopen = () => {
-            (this.term as any).attach(socket);
-            (this.term as any)._initialized = true;
-        };
-        socket.onclose = () => {
-            this.title.label = `<terminated>`;
-        };
-        socket.onerror = (err) => {
-            console.error(err);
-        };
-        this.toDispose.push(Disposable.create(() =>
-            socket.close()
-        ));
+        /* An error has occured in the backend.  */
+        if (!success) {
```

Comment:
I think await will result in a thrown error if the promise was rejected.

Reply:
this can't be rejected.